subsys/bluetooth/mesh/cfg.c

```diff
@@ -100,9 +100,11 @@ int bt_mesh_gatt_proxy_set(enum bt_mesh_feat_state gatt_proxy)
 		return err;
 	}
 
-	if (gatt_proxy == BT_MESH_FEATURE_DISABLED &&
-	    !bt_mesh_subnet_find(node_id_is_running, NULL)) {
-		bt_mesh_adv_gatt_update();
+	if ((IS_ENABLED(CONFIG_BT_EXT_ADV) &&
```

Comment:
Is this only related to extended advertiser or both advertisers has this issue?

Reply:
I tested it with both, and it seems like the legacy_adv manages without the fix.

Reply:
But if it does not cause any harm to perform the same procedure for legacy_adv it might be a little bit cleaner in the code?

Reply:
Yes, I think it should be independent of advertiser type.